contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -761,3 +761,38 @@ The central manager will poll each worker for their discovered volumes and creat
 them.  When a PV is released, then it will send the cleanup request to the worker.
 
 Detailed design TBD
+
+
+### Phase 3: 1.13 Alpha
+
+#### Dynamic Provisioning
+
+Note: The origin design that based on existing static discovery plugin is out-of-date. And currently the thing is blocked by the capcity report issue. Here I list the thoughts inflight, to avoid readers being misled by the old information. It's far from a complete deisgn, but more of rough thoughts on which we hope to receive feedback from readers.
+
+##### Overall Thoughts
+
+As we'll gradually switch all of the storages to CSI, we plan to construct a CSI driver for local storage, which will also be responsible for dynamically provisioning of local volumes. Users can still use the existing local storage plugin in external-storage repo to maintain local volumes in the "static way". The two plugins are completely independent.
+
+##### CSI Driver Details
+
+Different from other storages, we'll need to deploy a set of CSI provisioner on each of the nodes. We'll rely on feature volumeScheduling as a dependency. For local storage, the provisioner is only responsible for PVCs marked with "related plugin name + selected node annotation". That is, it only privision the pvcs that "scheduled" to its node.
```

Comment:
Will all these provisioners list/watch all PV/PVCs? If so, maybe we need to take care of the pressure on apiserver/etcd.

Reply:
The existing static discovery plugin also behavior like this. So I suppose this acceptable.